### ISSUE TITLE
session_active

### ISSUE THREAD

**Issue by aappling-usgs** [USGS-R/sbtools#137]
I'd like a single call to check whether I'm logged in to an active session. Right now the most concise I've found is `is.null(current_session()) || !session_validate()` - not awful, but a single function call would be nicer. Something like

```r
session_active <- function(session=current_session()) {
  !is.null(session) && session_validate(session)
}
```

It's possible something like this already exists in the package under a name I didn't recognize - if so, someone please chime in.

**Comment by jread-usgs** [comment]
:+1: from me

**Comment by aappling-usgs** [comment]
Now I think the above strategy might not be sufficient. I logged in more than an hour ago, let my R session sit for a while, then came back and ran an SB query. The query returned 673 results the first time and 0 just now, without any error resulting from `if(is.null(current_session()) || !session_validate()) stop()`. I logged in again and then got 673 back. I really would like to be able to send  an error or warning when the session goes inactive, but I clearly don't have the right code yet.

**Comment by sckott** [comment]
Hmm, is this the definitive docs on authentication for Sciencebase? https://my.usgs.gov/confluence/display/sciencebase/Authenticating

What about this: 

```r
is_session <- function(..., session = current_session()) {
	x <- GET(paste0(pkg.env$url_base, "jossoHelper/sessionInfo"), 
					 handle = session, ...)
	stop_for_status(x)
	content(x)$isLoggedIn
}
```

```r
is_session(config=verbose())
#> -> GET /catalog/jossoHelper/sessionInfo HTTP/1.1
#> -> Host: www.sciencebase.gov
#> -> User-Agent: libcurl/7.43.0 r-curl/0.9.4 httr/1.0.0
#> -> Accept-Encoding: gzip, deflate
#> -> Cookie: BNES_JSESSIONID=qktoyF7oOHGxDUjwUQjeUexHoMuSZC23r2WNwOa829BGb6HypEmkoaVLP8xZ7sNYgCpYPcUHhor+8y+lhnAjEvUmK2G4IAB6bwW5numm1JdO9zwwzW+vkw==; JSESSIONID=A6A1EFE89BCC3C840E0F14E9D75459A4
#> -> Accept: application/json, text/xml, application/xml, */*
#> -> 
#> <- HTTP/1.1 200 OK
#> <- Date: Wed, 23 Dec 2015 00:41:33 GMT
#> <- X-Frame-Options: DENY
#> <- X-UA-Compatible: IE=edge
#> <- Content-Type: application/json;charset=UTF-8
#> <- Transfer-Encoding: chunked
#> <- 
#> [1] TRUE
```

After restarting R session, and not re-authing:

```r
is_session()
#> [1] FALSE
```

thoughts?

**Comment by lawinslow** [comment]
Now available as `is_logged_in` as part of PR #146
